scripts/kconfig/kconfigfunctions.py

```diff
@@ -337,19 +333,38 @@ def dt_node_has_bool_prop(kconf, _, path, prop):
 
     return "n"
 
-def dt_node_has_prop(kconf, _, label, prop):
+def dt_node_has_bool_prop(kconf, _, path, prop):
     """
-    This function takes a 'label' and looks for an EDT node for that label. If
-    it finds an EDT node, it will look to see if that node has a property
+    This function takes a 'path' and looks for an EDT node at that path. If it
+    finds an EDT node, it will look to see if that node has a boolean property
     by the name of 'prop'.  If the 'prop' exists it will return "y" otherwise
     we return "n".
     """
+    if doc_mode or edt is None:
+        return "n"
+
+    return _dt_node_has_bool_prop_generic(edt.get_node, path, prop)
 
+def dt_nodelabel_has_bool_prop(kconf, _, label, prop):
+    """
+    This function takes a 'label' and looks for an EDT node with that label.
+    If it finds an EDT node, it will look to see if that node has a boolean
+    property by the name of 'prop'.  If the 'prop' exists it will return "y"
+    otherwise we return "n".
+    """
```

Comment:
Functions or API calls in the style of "has bool prop" are usually the wrong way to go.

For example, sometimes people will use [DT_NODE_HAS_PROP(foo, bar)](https://docs.zephyrproject.org/latest/reference/devicetree/api.html#group__devicetree-generic-exist_1gacce67bf20541cd2d07d8540058964692) to test a boolean property, but this should be [DT_PROP(foo, bar)](https://docs.zephyrproject.org/latest/reference/devicetree/api.html#group__devicetree-generic-prop_1ga8e1fd9ebacd85d2013df027d041d506b) instead. DT_PROP knows that the property is boolean because of the binding. Therefore, presence is true, and absence is false.

I have the same objection to this function here; this is semantically wrong because "having" a boolean property is the same thing as the property being true. So I'm not sure this is needed. This is different than dt_nodelabel_has_prop because some properties are optional. A boolean property is always optional, but its presence or absence have different semantics.

I would advocate for a `dt_nodelabel_bool_prop` function instead, which either returns y or n for boolean properties (and returns n by default). Then you can later on add `dt_nodelabel_int_prop_int`, `dt_nodelabel_int_prop_hex`, etc., and the naming will all be consistent.

Reply:
It sounds reasonable. You're right about the checking for boolean properties, I've used wrong functions too in the past.
Also the naming is more consistent with the int ones. Thanks, done that!